stage4.1_first_application/pycasa/ui/image_file_view.py

```diff
@@ -0,0 +1,33 @@
+# General imports
+from matplotlib.figure import Figure
+from matplotlib.pyplot import imshow, figure
+import imageio
+
+# ETS imports
+from traits.api import Instance, observe
+from traitsui.api import Item, ModelView, View
+
+# Local imports
+from ets_tutorial.util.mpl_figure_editor import MplFigureEditor
+from ..model.image_file import ImageFile
+
+
+class ImageFileView(ModelView):
+    """ ModelView for an image file object.
+    """
+    model = Instance(ImageFile)
+
+    figure = Instance(Figure)
+
+    view = View(
+        Item("model.filepath", style="readonly", show_label=False),
+        Item("figure", editor=MplFigureEditor(), show_label=False)
+    )
+
+    @observe("model.filepath")
+    def build_mpl_figure(self, event):
+        figure = Figure()
+        axes = figure.add_subplot(111)
+        data = imageio.imread(self.model.filepath)
+        axes.imshow(data)
```

Comment:
This could useful as a method on the model itself:

```python
# In ImageFile
def to_array(self):
    with PIL.Image.open(self.filepath) as img:
        return np.asarray(img)
```
And here:
```suggestion
        axes.imshow(self.model.to_array())
```

Reply:
Love it. Done.